From e78d74b3a776642367d6a0a3a45724c9f5220308 Mon Sep 17 00:00:00 2001
From: Eric Yang <Eric.Yang2@amd.com>
Date: Thu, 7 Sep 2017 14:53:23 -0400
Subject: [PATCH 4026/5855] drm/amd/display: fix usb-c dongle regression

A previous change moved link_disconnect_sink after dpcd read,
this causes the sink count to be overwritten which in turn causes
another detection to be triggered in the short pulse handler, which
brings down the display.

Signed-off-by: Eric Yang <Eric.Yang2@amd.com>
Reviewed-by: Tony Cheng <Tony.Cheng@amd.com>
Acked-by: Harry Wentland <Harry.Wentland@amd.com>
---
 drivers/gpu/drm/amd/display/dc/core/dc_link.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc_link.c b/drivers/gpu/drm/amd/display/dc/core/dc_link.c
index ee23b60..d5eaebe 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc_link.c
@@ -652,8 +652,6 @@ bool dc_link_detect(struct dc_link *link, enum dc_detect_reason reason)
 		if (link->local_sink)	{
 			sink = link->local_sink;
 		} else {
-			link_disconnect_sink(link);
-
 			sink_init_data.link = link;
 			sink_init_data.sink_signal = sink_caps.signal;
 
-- 
2.7.4

